File: ThematicMapCreator.Api/Models/Layer.cs
using System;

namespace ThematicMapCreator.Api.Models
{
    public class Layer
    {
        public string Data { get; set; }
        public Guid Id { get; set; }
        public int Index { get; set; }
        public Map Map { get; set; }
        public Guid MapId { get; set; }
        public string Name { get; set; }
        public string StyleOptions { get; set; }
        public LayerType Type { get; set; }
        public bool Visible { get; set; }
    }
}

File: ThematicMapCreator.Client/src/app/models/layer.ts
import { GeoJSON } from 'leaflet';
import { LayerType } from './layer-type.enum';
import { LayerStyleOptions } from './layer-style-options/layer-style-options';

export interface Layer {
    id: string;
    index: number;
    name: string;
    data: GeoJSON.GeoJsonObject;
    type: LayerType;
    styleOptions: LayerStyleOptions;
    visible: boolean;
    mapId: string;
}
